joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala

```diff
@@ -0,0 +1,85 @@
+package io.joern.rubysrc2cpg.astcreation
+
+import io.joern.rubysrc2cpg.parser.ParserAst.*
+import io.joern.rubysrc2cpg.parser.{ParserAst, ResourceManagedParser}
+import io.joern.rubysrc2cpg.passes.Defines
+import io.joern.x2cpg.datastructures.Stack.*
+import io.joern.x2cpg.{Ast, AstCreatorBase, AstNodeBuilder, ValidationMode}
+import io.shiftleft.codepropertygraph.generated.nodes.*
+import io.shiftleft.semanticcpg.language.types.structure.NamespaceTraversal
+import org.slf4j.{Logger, LoggerFactory}
+import overflowdb.BatchedUpdate
+
+import scala.util.{Failure, Success}
+
+class AstCreator(protected val filename: String, parser: ResourceManagedParser, projectRoot: Option[String] = None)(
+  implicit withSchemaValidation: ValidationMode
+) extends AstCreatorBase(filename)
+    with AstCreatorHelper
+    with AstForStatementsCreator
+    with AstForExpressionsCreator
+    with AstForFunctionsCreator
+    with AstForTypesCreator
+    with AstNodeBuilder[ParserNode, AstCreator] {
+
+  protected val logger: Logger = LoggerFactory.getLogger(getClass)
+
+  protected val relativeFileName: String =
+    projectRoot.map(filename.stripPrefix).map(_.stripPrefix(java.io.File.separator)).getOrElse(filename)
+
+  override def createAst(): BatchedUpdate.DiffGraphBuilder = {
+    parser.parse(filename) match
+      case Success(programCtx) =>
+        // TODO:
```

Comment:
What's the TODO here?

Reply:
Thanks for catching it! It was a leftover. Will remove it.